src/main/java/zd/zero/waifu/motivator/plugin/listeners/WaifuUnitTesterImpl.java

```diff
@@ -1,49 +1,37 @@
 package zd.zero.waifu.motivator.plugin.listeners;
 
-import com.intellij.execution.testframework.TestsUIUtil;
-import com.intellij.notification.Notification;
-import com.intellij.notification.NotificationType;
-import com.intellij.notification.Notifications;
+import com.intellij.execution.testframework.sm.runner.SMTRunnerEventsAdapter;
+import com.intellij.execution.testframework.sm.runner.SMTRunnerEventsListener;
+import com.intellij.execution.testframework.sm.runner.SMTestProxy;
 import com.intellij.util.messages.MessageBusConnection;
 import org.jetbrains.annotations.NotNull;
-import zd.zero.waifu.motivator.plugin.tools.Debouncer;
 
 public class WaifuUnitTesterImpl implements WaifuUnitTester {
 
     private final MessageBusConnection busConnection;
 
     private final WaifuUnitTester.Listener listener;
 
-    private final Debouncer debouncer;
-
     public WaifuUnitTesterImpl( MessageBusConnection busConnection,
-                                Listener listener,
-                                Debouncer debouncer ) {
+                                Listener listener ) {
         this.busConnection = busConnection;
         this.listener = listener;
-        this.debouncer = debouncer;
     }
 
     @Override
     public void init() {
-        busConnection.subscribe( Notifications.TOPIC, new Notifications() {
+        busConnection.subscribe( SMTRunnerEventsListener.TEST_STATUS, new SMTRunnerEventsAdapter() {
```

Comment:
Better topic to subscribe to, I still opt to programmatic topic subscription instead of a declarative one, as we're passing `project` to `WaifuUnitTester.Listener`

Reply:
If you know a way to instantiate `com.intellij.openapi.project.Project` through a utility class. I tried creating a [Light service](https://jetbrains.org/intellij/sdk/docs/basics/plugin_structure/plugin_services.html?search=servi#light-services) with the project level which should provide a `project` instance but having errors.

Reply:
`com.intellij.openapi.project.ProjectManager` has the ability to get the open projects and default projects, is that what you are looking for?

Reply:
I tried to use it but it can't give me the current project instance, maybe I'll just do it sometime soon as it'll affect many changes for this bug fix

Reply:
The `defaultProject` is not the current project?

Reply:
Unfortunately, it is not, and as described in the docs, it is just a template project, and not the current project instance opened.

> Returns the project which is used as a template for new projects. The template project is always available, even when no other project is open. This instance is not supposed to be used for anything except template settings storage.

Reply:
I guess that's what I get for not reading documentation 😅